src/main/java/teammates/common/util/AppUrl.java

```diff
@@ -0,0 +1,49 @@
+package teammates.common.util;
+
+/**
+ * A specific implementation of {@link Url} used to encapsulate URLs of the application.
+ */
+public class AppUrl extends Url {
+    
+    private static final String BASE_WEB_URL_REGEX = "^(https?://[^/]+)";
+    
+    public AppUrl(String baseUrl, String url) {
+        super(baseUrl, makeRelativeUrl(url)); // force the stored URL to be relative
+        Assumption.assertTrue(baseUrl.matches(BASE_WEB_URL_REGEX + "$"));
+        String relativeUrl = makeRelativeUrl(url);
+        Assumption.assertTrue(relativeUrl.isEmpty() || relativeUrl.startsWith("/"));
```

Comment:
1. Like I've said before, there is a use case for `file:///` links so I can't apply the assertion to the generic `Url` class. `file:///` links abstraction exists for the better, or the alternative is the 4 times repetition of `... new File(".").getCanonicalPath() ...` in `AllJsTests`, `AppPageUiTest`, `GodModeTest`, and `TableSortTest`, who knows if we're going to have more in the future.
2. Yes we can, but most of the time we're only supplying the relative URL. If we're supplying full URL all the time it defeats the purpose of suppressing repetition of base URL.